cmd/metacache-entries.go

```diff
@@ -335,10 +336,18 @@ func (m metaCacheEntries) resolve(r *metadataResolutionParams) (selected *metaCa
 		return selected, true
 	}
 
+	if selected.cached == nil {
+		_, err := selected.xlmeta()
+		if err != nil {
+			return nil, false
+		}
+	}
+
```

Comment:
Move this to after the `// If all objects agree.` return. If we don't need to decode, let's skip it.

Reply:
done